src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/world/entity/MixinEntity.java

```diff
@@ -0,0 +1,87 @@
+package io.github.opencubicchunks.cubicchunks.mixin.core.common.world.entity;
+
+import io.github.notstirred.dasm.api.annotations.Dasm;
+import io.github.notstirred.dasm.api.annotations.redirect.redirects.AddFieldToSets;
+import io.github.notstirred.dasm.api.annotations.redirect.redirects.AddTransformToSets;
+import io.github.notstirred.dasm.api.annotations.selector.FieldSig;
+import io.github.notstirred.dasm.api.annotations.selector.MethodSig;
+import io.github.notstirred.dasm.api.annotations.selector.Ref;
+import io.github.notstirred.dasm.api.annotations.transform.TransformFromMethod;
+import io.github.opencubicchunks.cc_core.utils.Coords;
+import io.github.opencubicchunks.cc_core.world.level.CloPos;
+import io.github.opencubicchunks.cubicchunks.CanBeCubic;
+import io.github.opencubicchunks.cubicchunks.mixin.dasmsets.ChunkToCubeSet;
+import io.github.opencubicchunks.cubicchunks.server.level.ServerCubeCache;
+import io.github.opencubicchunks.cubicchunks.world.entity.EntityCubePosGetter;
+import io.github.opencubicchunks.cubicchunks.world.level.CubicLevel;
+import io.github.opencubicchunks.cubicchunks.world.level.CubicLevelReader;
+import net.minecraft.core.BlockPos;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.server.level.TicketType;
+import net.minecraft.util.Mth;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.level.ChunkPos;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.AABB;
+import org.objectweb.asm.Opcodes;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+/**
+ *  We modify Entity to track its cube position, and to replace calls to chunk-specific methods with their corresponding cubic methods when in a cubic Level.
+ */
+@Dasm(ChunkToCubeSet.class)
+@Mixin(Entity.class)
+public abstract class MixinEntity implements EntityCubePosGetter {
+    @Shadow private Level level;
+    @Shadow private BlockPos blockPosition;
+
+    @Shadow public abstract void teleportTo(double x, double y, double z);
+    @Shadow public abstract AABB getBoundingBox();
+    @Shadow public abstract int getId();
+
+    @AddFieldToSets(sets = ChunkToCubeSet.class, owner = @Ref(Entity.class), field = @FieldSig(type = @Ref(ChunkPos.class), name = "chunkPosition"))
+    private CloPos cc_cubePosition = CloPos.cube(0, 0, 0);
+
+    @AddTransformToSets(ChunkToCubeSet.class) @TransformFromMethod(@MethodSig("chunkPosition()Lnet/minecraft/world/level/ChunkPos;"))
+    public native CloPos cc_cubePosition();
```

Comment:
Does it make sense for a user of `ChunkToCubeSet` to add a method returning `CloPos` to the `ChunkToCubeSet`?

I feel like this should be a `CubePos` and it converts to `CloPos` internally to add the ticket, it seems really weird for an entity to have a clo tbh.